### PR TITLE
feature/switch-to-jsonapi-for-you-might-like

### PR DESCRIPTION
**Pull request by jonhindle**
### Context

> Does this issue have a Trello card?

https://trello.com/c/P8Sx0rW5/210-switch-to-jsonapi-for-the-you-might-like-component

> If this is an issue, do we have steps to reproduce?

N/A

### Intent

> What changes are introduced by this PR that correspond to the above card?

Suggestions coming from JSONAPI, jsonAPI helper functions to reduce code duplication.

> Would this PR benefit from screenshots?

N/A

### Considerations

> Is there any additional information that would help when reviewing this PR?

N/A

> Are there any steps required when merging/deploying this PR?

N/A

### Checklist

- [ ] This PR contains **only** changes related to the above card
- [ ] Tests have been added/updated to cover the change
- [ ] Documentation has been updated where appropriate
- [ ] Tested in Development

### CODE REVIEW

File: server/repositories/cmsApi.js [hunk 1]
```diff
@@ -53,7 +53,7 @@ class CmsApi {
     try {
       return await call();
     } catch (err) {
-      if (err.response && statusCodes.includes(err.response.status)) {
+      if (statusCodes.includes(err?.response?.status)) {
```
Comment (andrewrlee) [inline server/repositories/cmsApi.js:56]: Fixed the commit merge to take your change. 
Only nitpicking comment is if err doesn't exist then the next line would blow up. 
Err should always be present according to the docs so NBD

File: server/services/cms.js [hunk 2]
```diff
@@ -65,64 +71,66 @@ class CmsService {
...
+              limit,
+            ),
+          )
+        : [];
+    return [...secondaryTagSuggestions, ...categorySuggestions]
+      .slice(0, limit)
+      .filter(
+        ({ id }, i, self) =>
+          self.findIndex(({ id: findId }) => id === findId) === i,
+      );
```
Comment (andrewrlee) [inline server/services/cms.js:98]: cannot read this, what's it doing?

Comment (andrewrlee) [reply server/services/cms.js:98]: wow, ok... think its removing duplicates. Maybe could extract into a named function or something?

File: server/services/cms.js [hunk 3]
```diff
@@ -65,64 +72,64 @@ class CmsService {
...
     );
+
     switch (type) {
       case 'node--page':
         return this.#cmsApi.get(new BasicPageQuery(location));
       case 'node--moj_pdf_item':
         return this.#cmsApi.get(new PdfPageQuery(location));
       case 'node--moj_radio_item':
-        return this.getAudio(establishmentName, establishmentId, location);
+        return this.getMedia(establishmentName, new AudioPageQuery(location));
```
Comment (andrewrlee) [inline server/services/cms.js:112]: 👍

**Review by davidthomas22 — APPROVED** [review]
LGTM

**Pull request merged**